test/integrationtests/username_password_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package integrationtests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/msal"
+)
+
+var labClientInstance = &labClient{}
+
+func AcquireTokenByUsernamePasswordCommonAAD(t *testing.T, user user, authority string, scope []string, appID string) {
+	options := msal.DefaultPublicClientApplicationOptions()
+	options.Authority = authority
+	publicClientApp, err := msal.NewPublicClientApplication(appID, &options)
+	if err != nil {
+		panic(err)
+	}
+	result, err := publicClientApp.AcquireTokenByUsernamePassword(
+		context.Background(),
+		scope,
+		user.Upn,
+		user.Password,
+	)
+	if err != nil {
+		t.Error(err)
+	}
+	if result.AccessToken == "" {
+		t.Error("No access token found")
+	}
+	if result.Account.GetUsername() != user.Upn {
+		t.Errorf("Incorrect user account")
+	}
+}
+func TestAcquireTokenByUsernamePasswordManaged(t *testing.T) {
```

Comment:
These 4 tests should be one table driven test.  If you haven't seen a table driven tests, if you do a google search for "golang table driven test" several articles should come up